pallets/external-validators/src/mock.rs

```diff
@@ -0,0 +1,341 @@
+    },
+};
+
+type Block = frame_system::mocking::MockBlock<Test>;
+
+// Configure a mock runtime to test the pallet.
+frame_support::construct_runtime!(
+    pub enum Test
+    {
+        System: frame_system,
+        ExternalValidators: pallet_external_validators,
+        Session: pallet_session,
+        Balances: pallet_balances,
+        Timestamp: pallet_timestamp,
+        Mock: mock_data,
+    }
+);
+
+parameter_types! {
+    pub const BlockHashCount: u64 = 250;
+    pub const SS58Prefix: u8 = 42;
+}
+
+impl system::Config for Test {
+    type BaseCallFilter = frame_support::traits::Everything;
+    type BlockWeights = ();
+    type BlockLength = ();
+    type DbWeight = ();
+    type RuntimeOrigin = RuntimeOrigin;
+    type RuntimeCall = RuntimeCall;
+    type Hash = H256;
+    type Hashing = BlakeTwo256;
+    type AccountId = u64;
+    type Lookup = IdentityLookup<Self::AccountId>;
+    type RuntimeEvent = RuntimeEvent;
+    type BlockHashCount = BlockHashCount;
+    type Version = ();
+    type PalletInfo = PalletInfo;
+    type AccountData = AccountData<u64>;
+    type OnNewAccount = ();
+    type OnKilledAccount = ();
+    type SystemWeightInfo = ();
+    type SS58Prefix = SS58Prefix;
+    type OnSetCode = ();
+    type MaxConsumers = frame_support::traits::ConstU32<16>;
+    type Nonce = u64;
+    type Block = Block;
+    type RuntimeTask = ();
+    type SingleBlockMigrations = ();
+    type MultiBlockMigrator = ();
+    type PreInherents = ();
+    type PostInherents = ();
+    type PostTransactions = ();
+}
+
+parameter_types! {
+    pub const ExistentialDeposit: u64 = 5;
+    pub const MaxReserves: u32 = 50;
+}
+
+impl pallet_balances::Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type WeightInfo = ();
+    type Balance = u64;
+    type DustRemoval = ();
+    type ExistentialDeposit = ExistentialDeposit;
+    type AccountStore = System;
+    type ReserveIdentifier = [u8; 8];
+    type RuntimeHoldReason = ();
+    type RuntimeFreezeReason = ();
+    type FreezeIdentifier = ();
+    type MaxLocks = ();
+    type MaxReserves = MaxReserves;
+    type MaxFreezes = ConstU32<0>;
+}
+
+impl pallet_timestamp::Config for Test {
+    type Moment = u64;
+    type OnTimestampSet = ();
+    type MinimumPeriod = ConstU64<5>;
+    type WeightInfo = ();
+}
+
+ord_parameter_types! {
+    pub const RootAccount: u64 = 777;
+}
+
+pub struct IsRegistered;
+impl ValidatorRegistration<u64> for IsRegistered {
+    fn is_registered(id: &u64) -> bool {
+        *id != 42u64
+    }
+}
+
+parameter_types! {
+    pub const SessionsPerEra: SessionIndex = 6;
+}
+
+impl Config for Test {
+    type RuntimeEvent = RuntimeEvent;
+    type UpdateOrigin = EnsureSignedBy<RootAccount, u64>;
+    type HistoryDepth = ConstU32<84>;
+    type MaxWhitelistedValidators = ConstU32<20>;
+    type MaxExternalValidators = ConstU32<20>;
+    type ValidatorId = <Self as frame_system::Config>::AccountId;
+    type ValidatorIdOf = IdentityCollator;
```

Comment:
I think you can put directly:
`type ValidatorIdOf = ConvertInto;`

Reply:
*fixed*